Actor.cs

```diff
@@ -0,0 +1,85 @@
+internal class Actor : MonoBehaviour
+{
+	private static List<Actor> _actors = new List<Actor>();
+	public int StartingHealth;
+	int _currentHealth;
+	private bool _isDead;
+	private CancellationTokenSource _healthCancellationTokenSource;
+	private CancellationTokenSource _deathCancellationTokenSource;
+
+	public event Action<Actor> ActorDied;
+	public event Action<Actor> ActorHealthChanged;
+
+	public void TakeDamage(int damage)
+	{
+		_currentHealth -= damage;
+		ActorHealthChanged?.Invoke(this);
+		RecordHealthToServer();
+		
+		if (_currentHealth <= 0)
+		{
+			_isDead = true;
+			ActorDied?.Invoke(this);
+			RecordDeathToServer();
+		}
+	}
+
+	private void Start()
+	{
+		_currentHealth = StartingHealth;
```

Comment:
This is probably more fitting in Awake than start?

If we run it in Start & we instantiate the object to a disabled parent start will never run until we enable it
We could also be doing health checks directly after instantiation which would then fail because their `_currentHeath` variable is set to default(int)